src/Voron/Data/Tables/NewPageAllocator.cs

```diff
@@ -199,6 +199,8 @@ public unsafe Page AllocateSinglePage(long nearbyPage)
                         {
                             if (TryMoveNextCyclic(it, startPage) == false)
                                 break;
+
```

Comment:
If I understand the code properly, we're looking for the first lower bit in the buffer. In the first `for` loop, when we find such an `ulong`, we are indicating that there is free space. Can we optimize the code and leverage the fact that we partially know where it is? I think we can eliminate the second `for` loop, which is iterating over the whole bitmap.

cc @redknightlois

Reply:
https://issues.hibernatingrhinos.com/issue/RavenDB-24100/Optimize-AllocateSinglePage

Reply:
yeah, but as part of a different issue :)
https://issues.hibernatingrhinos.com/issue/RavenDB-24100/Optimize-AllocateSinglePage

Reply:
Yes, that is `popcount` dependent and probably we could do some other things in there. If you need any help on that loop me in.